src/patterns/ec2-app.ts

```diff
@@ -9,17 +11,45 @@ import type { Gu5xxPercentageMonitoringProps, NoMonitoring } from "../constructs
 import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { GuStack } from "../constructs/core";
 import { AppIdentity } from "../constructs/core/identity";
-import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuPublicInternetAccessSecurityGroup, GuSecurityGroup, GuVpc, SubnetType } from "../constructs/ec2";
 import { GuGetPrivateConfigPolicy, GuInstanceRole } from "../constructs/iam";
 import {
   GuApplicationLoadBalancer,
   GuApplicationTargetGroup,
   GuHttpsApplicationListener,
 } from "../constructs/loadbalancing";
 
+export enum AccessScope {
+  PUBLIC,
+  RESTRICTED,
+}
+
+export interface Access {
+  scope: AccessScope;
+}
+
+/*
+ * For when you want your application to be accessible to the world (0.0.0.0/0).
+ * Your application load balancer will have a public IP address that can be reached by anyone,
+ * so only use if you are aware and happy with the consequences!
+ * */
+interface PublicAccess extends Access {
+  scope: AccessScope.PUBLIC;
+}
+
+/*
+ * For when you want to restrict your application's access to a list of CIDR ranges.
+ * */
+interface RestrictedAccess extends Access {
+  scope: AccessScope.RESTRICTED;
+  cidrRanges: IPeer[];
+}
+
+export type AppAccess = PublicAccess | RestrictedAccess;
+
 interface GuEc2AppProps extends AppIdentity {
   userData: GuUserDataProps | string;
-  publicFacing: boolean; // could also name it `internetFacing` to match GuApplicationLoadBalancer
+  access?: AppAccess; // If not specifying an access type, we can assume it to be open to the world
```

Comment:
Minor: personally I quite like the explicitness of a required prop for something like this.

Reply:
Thinking about it, I think you're right. I've changed it to be mandatory for now, I think we can address sensible defaults (if any) in future PRs as it might require some more careful thought.